lib/splitclient-rb/engine/matchers/matcher.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  #
+  # class to implement the all keys matcher
+  #
+  class Matcher
+    #
+    # evaluates if the given object equals the matcher
+    #
+    # @param obj [object] object to be evaluated
+    #
+    # @return [boolean] true if obj equals the matcher
+    def equals?(obj)
+      if obj.nil?
+        false
+      elsif !obj.instance_of?(self.class)
+        false
+      elsif equal?(obj)
```

Comment:
this makes the rest of the method useless since `equal?` will test if `obj` and `self` are the same if they are the same they won't be true for the other cases, and if they are not equal (by being nil, a different class, or something else) they will go to the `else` case anyway